### DISCUSSION TITLE
Sort order for the app index page of databases

### DISCUSSION THREAD

**Discussion by justinallen** [simonw/datasette#1021] [General]
Apologies if this has been answered elsewhere, but it's a little mysterious to me what determines the sort order of available databases on the app index page when running in configuration directory mode. They don't seem to load either in alphanumeric order, or the order added, or the order opened! But maybe I am missing something obvious? Is there, or could there be, a way to determine this sort order as a configuration option in the metadata file?

**Comment by simonw** [comment]
Oh that's a good question. I think that order may be undefined - if you pass explicit database files (`datasette one.db two.db three.db`) they will be shown in that order, but for configuration directory mode it's whatever order they come back in from `glob("*.db"):

https://github.com/simonw/datasette/blob/4f7c0ebd85ccd8c1853d7aa0147628f7c1b749cc/datasette/app.py#L200-L201

Letting `metadata.json/yml` over-ride that order is one option. I also have open tickets to redesign that page (#991) and to add search and pagination (#461).

I'd love suggestions on how this index page could be more useful!

**Reply by justinallen** [reply]
Thanks for the reply, now that's clear.

**Comment by simonw** [comment]
How many databases do you have attached? Do you want to control the exact order, or would you be happy highlighting a subset of them in some way?

I was looking at the GitHub Discussions homepage and how it lets you highlight specific threads and thinking that might be an interesting option for the Datasette instance homepage:

<img width="1402" alt="Banners_and_Alerts_and_Discussions_·_simonw_datasette" src="https://user-images.githubusercontent.com/9599/96322111-b0452a80-0fcc-11eb-80fa-a3cf7cf473b3.png">

I'm also thinking that maybe the focus of the homepage should actually be more on the tables than the databases themselves. Tables are by far the most important content in any Datasette instance - the fact that they are organized into databases is I think of secondary importance.

**Reply by justinallen** [reply]
Presently there's just a few databases each with a few tables, but we're thinking about using a Datasette instance as an archive and resource for reporters which could expand quite a bit - dozens of databases, hundreds of tables? Maybe! It makes sense that the tables are the main thing and should be prominent - and the ability to highlight or pin certain databases also makes sense. I'm thinking of the tables and databases a little like files and directories for keeping things organized and also, grouping data which we'd want to query across tables.
